machine/machine.go

```diff
@@ -313,3 +313,57 @@ func getNumCores() int {
 
 	return numCPU
 }
+
+var cpuToSocket map[int]int
+
+// GetSocketFromCPU returns socket which is bounded to passed CPU.
+// If there is no information it would return -1.
+func GetSocketFromCPU(cpu int) int {
+	// Check if information is cached.
+	if cpuToSocket == nil {
+		var err error
+		cpuToSocket, err = readCPUToSocket(cpuBusPath)
```

Comment:
Please write a unit tests to cover the case when `physical_package_id` file is not available.

Reply:
Functionality moved to ``utils/sysinfo/GetSocketFromCPU``.